dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/constant/ConstantServiceTest.java

```diff
@@ -0,0 +1,288 @@
+        assertEquals( constantA, constantMap.get( constantA.getUid() ) );
+        assertEquals( constantB, constantMap.get( constantB.getUid() ) );
+    }
+
+    @Test
+    public void testGetConstantParameterMap()
+    {
+        constantService.saveConstant( constantA );
+        constantService.saveConstant( constantB );
+
+        Map<String, Double> constantParameterMap = constantService.getConstantParameterMap();
+
+        assertEquals( 2, constantParameterMap.size() );
+        assertEquals( 12.0, constantParameterMap.get( constantA.getName() ), .001 );
+        assertEquals( 13.0, constantParameterMap.get( constantB.getName() ), .001 );
+    }
+
+    @Test
+    public void testGetConstantCount()
+    {
+        constantService.saveConstant( constantA );
+        constantService.saveConstant( constantB );
+
+        assertEquals( 2, constantService.getConstantCount() );
+    }
+
+    @Test
+    public void testGetConstantCountByName()
+    {
+        constantService.saveConstant( constantA );
+        constantService.saveConstant( constantB );
+        constantService.saveConstant( constantB2 );
+
+        assertEquals( 3, constantService.getConstantCountByName( "A" ) );
+        assertEquals( 2, constantService.getConstantCountByName( "B" ) );
+        assertEquals( 1, constantService.getConstantCountByName( "2" ) );
+    }
+
+    @Test
+    public void testGetConstantsBetween()
+    {
+        constantService.saveConstant( constantA );
+        constantService.saveConstant( constantB );
+        constantService.saveConstant( constantB2 );
+
+        List<Constant> constants = constantService.getConstantsBetween( 0, 2 );
+
+        assertEquals( 2, constants.size() );
+        assertTrue( constants.contains( constantA ) );
+        assertTrue( constants.contains( constantB ) );
+
+        constants = constantService.getConstantsBetween( 0, 3 );
+
+        assertEquals( 3, constants.size() );
+        assertTrue( constants.contains( constantA ) );
+        assertTrue( constants.contains( constantB ) );
+        assertTrue( constants.contains( constantB2 ) );
+
+        constants = constantService.getConstantsBetween( 1, 3 );
+
+        assertEquals( 2, constants.size() );
+        assertTrue( constants.contains( constantB ) );
+        assertTrue( constants.contains( constantB2 ) );
+    }
+
+    public void testGetConstantsBetweenByName()
+    {
+        constantService.saveConstant( constantA );
+        constantService.saveConstant( constantB );
+        constantService.saveConstant( constantB2 );
+        constantService.saveConstant( constantB3 );
+
+        List<Constant> constants = constantService.getConstantsBetweenByName( "A", 0, 3 );
+
+        assertEquals( 4, constants.size() );
+        assertTrue( constants.contains( constantA ) );
+        assertTrue( constants.contains( constantB ) );
+        assertTrue( constants.contains( constantB2 ) );
+        assertTrue( constants.contains( constantB3 ) );
```

Comment:
Better to create a helper method to assert expected sets, like
```java
void assertHasConstants(String name, int first, int max, Constant... expected) 
```
so these blocks don't have to be duplicated and make the test lengthy.

Reply:
Nice idea! We should put that in a common place for tests.